maildir/maildir.c

```diff
@@ -507,7 +507,10 @@ int maildir_mbox_check(struct Mailbox *m, int *index_hint)
   /* Incorporate new messages */
   num_new = maildir_move_to_mailbox(m, &md);
   if (num_new > 0)
+  {
     mutt_mailbox_changed(m, MBN_INVALID);
+    m->changed = true;
```

Comment:
I feel like it makes more to put `m->changed = true;` inside `mutt_mailbox_changed(...)`. Otherwise, `mutt_mailbox_changed(...)` seems like an inaccurate name.

Reply:
The name `mutt_mailbox_changed()` isn't a great one -- `mutt_mailbox_notify()` might be better.
I didn't want to move `m->changed = true` because I didn't want to change the behaviour of the other callers -- which may, or may not, be correct :-)

Ultimately, `mutt_mailbox_changed()` is a temporary fix and will die soon
("soon" == when I stop making small changes and finish the big ones).